### PR TITLE
DT-22452 Find Forms UI upgrade

### PR DESCRIPTION
**Pull request by zacharymorel**
## Issue
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22452

## Description
Make Find Forms appearance consistent in an effort that will reach across all search tools on va.gov.

## Testing done
Update Unit tests, ran unit and e2e, spun up local env. 

## Screenshots
Screen Shot 2021-06-07 at 11 19 30 AM
Screen Shot 2021-06-07 at 11 20 33 AM

## Acceptance criteria
- [x] Find Forms search tool UI is updated to mockup.
- [x] Find Forms search tool UX is the same as it is currently.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/components/FormTitle.jsx [hunk 1]
```diff
@@ -3,27 +3,31 @@ import PropTypes from 'prop-types';
...
+      <>
+        <p className="vads-u-font-weight--normal vads-u-margin--0">Form {id}</p>
+        <a
+          href={formUrl}
+          className="vads-u-text-decoration--none vads-u-margin-top--1"
+          onClick={() => recordGAEvent(title, formUrl, 'title')}
+          lang={lang}
+        >
+          {title}{' '}
+          <i
```
[the remark this answers is not in the document]
Comment (zacharymorel) [reply src/applications/find-forms/components/FormTitle.jsx:19]: I'm not sure why this is here. 
Is there background to the Icon bot?

[the remark this answers is not in the document]
Comment (alexandec) [reply src/applications/find-forms/components/FormTitle.jsx:19]: The idea behind this check is that purely decorative icons should be have `aria-hidden=true` so they are ignored by screen readers. This prevents confusion for screen reader users. Here's some more background on icons and a11y: https://css-tricks.com/can-make-icon-system-accessible/

[the remark this answers is not in the document]
Comment (zacharymorel) [reply src/applications/find-forms/components/FormTitle.jsx:19]: Ah so it's a reminder just before the code gets merged.

**Review by alexandec — APPROVED** [review]

**Comment by zacharymorel** [comment]
@kelsonic can you give this a look over. 

Note, **useSearchUIUXEnhancements** is the new code, in each component, anything after the ternary is the old code so stakeholders can review on staging.

File: src/applications/find-forms/components/FormTitle.jsx [hunk 2]
```diff
@@ -1,39 +1,87 @@
...
+          </a>
+        ) : (
+          <>
+            {id} {title}
+          </>
+        )}
+      </dt>
     )}
-  </dt>
+  </>
```
Comment (kelsonic) [inline src/applications/find-forms/components/FormTitle.jsx:76]: I think we can omit the `<>` since it only has 1 child if you want 🙂

**Review by kelsonic — APPROVED** [review]
lgtm! 🎉

**Pull request merged**